sqlglot/generator.py

```diff
@@ -218,6 +218,15 @@ def sep(self, sep=" "):
     def seg(self, sql, sep=" "):
         return f"{self.sep(sep)}{sql}"
 
+    def maybe_comment(self, sql, comment):
+        if not self._comments or not comment:
+            return sql
+
+        if not self.pretty:
+            return f"{sql}{f' /* {comment} */'}"
```

Comment:
i think the comment start and comment end should be passed in because different dialects could override it

basically find a comment with start and end, and one without there are two different types

default can be -- and /*

Reply:
or if you do some research and find everybody supports -- and /* then we don't need to add it

Reply:
The dialects we currently have in sqlglot seem to support this syntax so we can omit the extra arguments for now I think.